akka-actor-typed-tests/src/test/scala/akka/actor/typed/ActorContextSpec.scala

```diff
@@ -1,867 +1,603 @@
-        startWith
-          .mkChild(Some("B"), ctx.spawnMessageAdapter(ChildEvent), self)
-          .stimulate(_._1 ! GetChild("A", self), _ ⇒ Child(None))
-          .stimulate(_._1 ! GetChild("B", self), x ⇒ Child(Some(x._2)))
-          .stimulate(_._1 ! GetChildren(self), x ⇒ Children(Set(x._2)))
-      })
+      type Children = Seq[ActorRef[Nothing]]
+      val probe = TestProbe[Children]()
+      val actor = spawn(Behaviors.immutablePartial[String] {
+        case (ctx, "B") ⇒
+          ctx.spawn(Behaviors.empty, "B")
+          probe.ref ! ctx.child("B").toSeq
+          Behaviors.same
+        case (ctx, "all") ⇒
+          probe.ref ! ctx.children.toSeq
+          Behaviors.same
+        case (ctx, name) ⇒
+          probe.ref ! ctx.child(name).toSeq
+          Behaviors.same
+      }.decorate)
+      actor ! "B"
+      val children = probe.expectMessageType[Children]
+      actor ! "A"
+      probe.expectMessage(Seq.empty)
+      actor ! "all"
+      probe.expectMessage(children)
+      children.size shouldBe 1
+      children.head.path.name shouldBe "B"
     }
 
     "set small receive timeout" in {
-      sync(setup("ctx30") { (ctx, startWith) ⇒
-        val self = ctx.self
-        startWith
-          .stimulate(_ ! SetTimeout(1.nano, self), _ ⇒ TimeoutSet)
-          .expectMessage(expectTimeout) { (msg, _) ⇒
-            msg should ===(GotReceiveTimeout)
-          }
-      })
+      val probe = TestProbe[Event]()
+      val actor = spawn(Behaviors.immutablePartial[Command] {
+        case (_, ReceiveTimeout) ⇒
+          probe.ref ! GotReceiveTimeout
+          Behaviors.same
+        case (ctx, SetTimeout(duration)) ⇒
+          ctx.setReceiveTimeout(duration, ReceiveTimeout)
+          probe.ref ! TimeoutSet
+          Behaviors.same
+      }.decorate)
+      actor ! SetTimeout(1.nano)
+      probe.expectMessage(TimeoutSet)
+      probe.expectMessage(GotReceiveTimeout)
     }
 
     "set large receive timeout" in {
-      sync(setup("ctx31") { (ctx, startWith) ⇒
-        val self = ctx.self
-        startWith
-          .stimulate(_ ! SetTimeout(1.minute, self), _ ⇒ TimeoutSet)
-          .stimulate(_ ⇒ ctx.schedule(1.second, self, Pong2), _ ⇒ Pong2)
-          .stimulate(_ ! Ping(self), _ ⇒ Pong1)
-
-      })
+      val probe = TestProbe[Event]()
+      val actor = spawn(Behaviors.immutablePartial[Command] {
+        case (ctx, Inert) ⇒
+          ctx.schedule(1.second, probe.ref, InertEvent)
+          Behaviors.same
+        case (_, Ping) ⇒
+          probe.ref ! Pong
+          Behaviors.same
+        case (_, ReceiveTimeout) ⇒
+          probe.ref ! GotReceiveTimeout
+          Behaviors.same
+        case (ctx, SetTimeout(duration)) ⇒
+          ctx.setReceiveTimeout(duration, ReceiveTimeout)
+          probe.ref ! TimeoutSet
+          Behaviors.same
+      }.decorate)
+      actor ! SetTimeout(1.minute)
+      probe.expectMessage(TimeoutSet)
+      actor ! Inert
+      probe.expectMessage(InertEvent)
```

Comment:
This doesn't really match what "inert" means, maybe use a custom protocol (strings maybe even) for this test case to make it easier to read